aws/data_source_aws_ec2_instance_type.go

```diff
@@ -0,0 +1,439 @@
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_gpu_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"fpgas": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"memory_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_fpga_memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"supported_placement_strategies": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"accelerators": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"manufacturer": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"hibernation_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"burstable_performance_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"dedicated_hosts_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"auto_recovery_supported": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEc2InstanceTypeRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	params := &ec2.DescribeInstanceTypesInput{}
+
+	instanceType := d.Get("instance_type").(string)
+	params.InstanceTypes = []*string{aws.String(instanceType)}
+	log.Printf("[DEBUG] Reading instances types: %s", params)
+
+	resp, err := conn.DescribeInstanceTypes(params)
+	if err != nil {
+		return err
+	}
+	if len(resp.InstanceTypes) == 0 {
+		return fmt.Errorf("no Instance Type found for %s", instanceType)
+	}
+	if len(resp.InstanceTypes) > 1 {
+		return fmt.Errorf("multiple instance types found for type %s", instanceType)
+	}
+	v := resp.InstanceTypes[0]
+	d.Set("instance_type", v.InstanceType)
+	if err := d.Set("current_generation", aws.BoolValue(v.CurrentGeneration)); err != nil {
+		return fmt.Errorf("error setting current_generation: %s", err)
+	}
```

Comment:
From lines 274-436, please simplify all the `d.Set()` statements like this example:
```suggestion
	d.Set("current_generation", v.CurrentGeneration)
```